### PR TITLE
Sync Mirage Setup

### PR DESCRIPTION
**Pull request by zofskeez**
This PR handles the initial mirage setup for sync endpoints based on the API documentation.

### CODE REVIEW

File: ui/mirage/factories/sync-association.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { Factory } from 'ember-cli-mirage';
+
+export default Factory.extend({
+  accessor: 'kv_eb4acbae',
+  secret_name: 'my-secret-1',
```
Comment (hellobontempo) [inline ui/mirage/factories/sync-association.js:10]: Do we want to make this a `nested/secret` ? Since most secrets in kv are nested

Comment (zofskeez) [reply ui/mirage/factories/sync-association.js:10]: I just copied the example from the docs as the default. The way it is set up is the `secret_name` param will be used when setting an association which will be the value selected in the UI. I'm happy to update the default though to something like `my-path/my-secret-1` for example.

File: ui/mirage/scenarios/sync.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+export default function (server) {
+  const types = ['aws-sm', 'azure-kv', 'gcp-sm', 'gh', 'vercel-project'];
```
Comment (hellobontempo) [inline ui/mirage/scenarios/sync.js:7]: Once this is merged, I can update this to use the helper I made!

**Review by hellobontempo — APPROVED** [review]
Thanks for setting this up!

File: ui/mirage/scenarios/sync.js (hunk 2 above)
Comment (zofskeez) [reply ui/mirage/scenarios/sync.js:7]: Nice ya I was going to leave a comment about that but you didn't need it 😁

**Pull request merged**